internal/impl/pure/scanner_csv_test.go

```diff
@@ -81,3 +84,111 @@ a4,b4,c4
 		`["a4","b4","c4"]`,
 	)
 }
+
+func TestCSVScannerExpectedHeaders(t *testing.T) {
+	confSpec := service.NewConfigSpec().Field(service.NewScannerField("test"))
+	pConf, err := confSpec.ParseYAML(`
+test:
+  csv:
+    expected_headers:
+      - a
+      - b
+      - c
+`, nil)
+	require.NoError(t, err)
+
+	rdr, err := pConf.FieldScanner("test")
+	require.NoError(t, err)
+
+	testutil.ScannerTestSuite(t, rdr, nil, []byte(`a,b,c
+a1,b1,c1
+a2,b2,c2
+a3,b3,c3
+a4,b4,c4
+`),
```

Comment:
mega nit: you can make these multi-line strings more readable by adding a newline and cutting it off on evaluation

```suggestion
	testutil.ScannerTestSuite(t, rdr, nil, []byte([1:]`
a,b,c
a1,b1,c1
a2,b2,c2
a3,b3,c3
a4,b4,c4
`),
```

Reply:
Don't think that will work